pheps/phep-0003.md

```diff
@@ -0,0 +1,152 @@
+
+## Code to generate support and drop schedules:
+```python
+import pandas as pd
+from datetime import datetime, timedelta
+
+# Define the release dates of the Python versions
+py_releases = {
+    "3.9": "Oct 5, 2020",
+    "3.10": "Oct 4, 2021",
+    "3.11": "Oct 24, 2022",
+    "3.12": "Oct 2, 2023",
+}
+
+# Define the support periods of 3.5 years / 6 months
+plus42 = timedelta(days=int(365*3.5 + 1))
+plus6 = timedelta(days=int(365 * 0.5))
+
+# Create a DataFrame from the predefined py_releases data
+data = []
+for version, release_date in py_releases.items():
+    release_date_dt = datetime.strptime(release_date, "%b %d, %Y")
+    drop_date_dt = release_date_dt + plus42
+    support_by_dt = release_date_dt + plus6
+    data.append((version, release_date_dt, support_by_dt, drop_date_dt))
+
+# Create a DataFrame
+df = pd.DataFrame(data, columns=["python version", "release", "support by", "drop"])
+df["quarter"] = df["drop"].dt.to_period("Q")
+
+# Save the DataFrame to a CSV file
+df.to_csv('python_version_support_schedule.csv', index=False)
+
+# Generate Mermaid syntax for the Gantt chart
+mermaid_gantt = """gantt
+    dateFormat  YYYY-MM-DD
+    title       Python Version Support Window
+    axisFormat  %m / %Y
+
+    section python
+"""
+
+for _, row in df.iterrows():
+    mermaid_gantt += f"    {row['python version']} : {row['release'].strftime('%Y-%m-%d')},{row['drop'].strftime('%Y-%m-%d')}\n"
+
+# Save the Mermaid syntax to a Markdown file
+# The Mermaid Gantt chart syntax is saved in 'python_version_support_chart.md'.
+# You can paste the content of that file into https://mermaid.live/ to generate the chart image.
+with open("python_version_support_chart.md", "w") as file:
+    file.write(mermaid_gantt)
+
+print("Version support schedule saved to 'python_version_support_schedule.csv'")
+print("Mermaid Gantt chart syntax saved to 'python_version_support_chart.md' (render at https://mermaid.live/)")
+```
+
+# Rejected Ideas
+<a name="rejected-ideas"></a>
+[SPEC-0](https://scientific-python.org/specs/spec-0000/)'s 36-month support timeline was considered instead of [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html)'s 42 months, but it was ultimately decided that the extra stability of 42 months is more desirable.
+
+# Open Issues
+<a name="open-issues"></a>
+1. This PHEP recommends supporting Python versions for 42 months (3.5 years) which is in line with [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html), _not_ 36 months (3 years) as recommended by [SPEC-0](https://scientific-python.org/specs/spec-0000/). The thinking is that 42 months is more lenient and therefore less of a burden to the community, but is that the right call? Should it be 36 months instead?
```

Comment:
On the numpy page it says:

> This NEP is superseded by the scientific python ecosystem coordination guideline [SPEC 0 — Minimum Supported Versions](https://scientific-python.org/specs/spec-0000/).

So maybe we should adopt SPEC-0 instead?

Reply:
Wow, I missed that entirely! And I go to the NEP 29 page kind of a lot!  I agree we should now go with SPEC 0 instead of NEP 29.